lib/db.js

```diff
@@ -92,24 +92,45 @@ class LimitDBRedis extends EventEmitter {
     this.buckets[key] = utils.buildBucket(bucket);
   }
 
-  bucketKeyConfig(type, key) {
-    const fromCache = type.overridesCache.get(key);
+  paramsToBucketConfig(params) {
+    const bucketConfig = {};
+
+    if (params && params.configOverride) {
+      if (params.configOverride.bucketSize) {
+        bucketConfig.size = params.configOverride.bucketSize;
+      }
+      if (params.configOverride.refillRateMsPerToken) {
+        bucketConfig.per_interval = 1;
+        bucketConfig.interval = params.configOverride.refillRateMsPerToken;
+      }
+    }
+
+    return bucketConfig;
+  }
+
+  /**
+   * @param {string} type
+   * @param {object} params
+   * @returns
+   */
+  bucketKeyConfig(type, params) {
+    const fromCache = type.overridesCache.get(params.key);
 
     if (fromCache) {
-      return fromCache;
+      return _.merge(fromCache, this.paramsToBucketConfig(params));
```

Comment:
Minor: `{ ...fromCache, ...this.paramsToBucketConfig(params) }` should work here; I prefer the native option vs lodash one when available.

Reply:
Agree, that's a better move.  Just gotta break some lodash muscle memory...